include/bouncer.h

```diff
@@ -567,6 +605,19 @@ struct PgSocket {
 
 	VarCache vars;		/* state of interesting server parameters */
 
+	PgClientPreparedStatement *client_prepared_statements;
+	PgServerPreparedStatement *server_prepared_statements;
+
+	struct StatList outstanding_requests;
```

Comment:
Currently the outstanding requests are stored at the server. Should they be stored at the client instead? It seems like there could be arguments for both.

Reply:
Do I correctly understand that the only difference between storing outstanding requests in client rather than in server is error handling: what to do with this requests when client or server is disconnected?

Reply:
Yeah I think that's the main difference. And I guess wherever we store them we could do the same, e.g. even if we store them at the server. It probably makes sense to clear them when the client disconnects.

The other difference is mainly cosmetical: i.e.